libfreerdp/crypto/per.c

```diff
@@ -352,11 +353,12 @@ BOOL per_read_object_identifier(wStream* s, BYTE oid[6])
  * Write PER OBJECT_IDENTIFIER (OID)
  * @param s stream
  * @param oid object identifier (oid)
+ * @warning It works correctly only for limited set of OIDs.
  */
 
 void per_write_object_identifier(wStream* s, BYTE oid[6])
 {
-	BYTE t12 = (oid[0] << 4) & (oid[1] & 0x0F);
+	BYTE t12 = oid[0] * 40 + oid[1];
```

Comment:
Where do you get the change from the shift by 4 (division by 16) to division by 40?